packages/stream-tile-handler/src/tile-document-handler.ts

```diff
@@ -210,8 +217,41 @@ export class TileDocumentHandler implements StreamHandler<TileDocument> {
   async _verifySignature(
     commitData: CommitData,
     context: Context,
-    controller: string
+    controller: string,
+    streamId?: StreamID
```

Comment:
why is streamId optional?  When would you want to call `_verifySignature` without the streamid?

Reply:
when its not a signed commit we dont need it, as it cannot have a cacao attached

Reply:
If it's not signed, why would we be calling `_verifySignature` at all?

Reply:
looking closer, it seems like the one place we call this without passing a streamid is for validating signed genesis commits.  So I guess it's about whether and how we support capabilities that grant the ability to create new streams on the user's behalf.  While we could calculate the StreamID in `_applyGenesis`, I doubt we'll have capabilities that grant the right to create a new Stream but only if the new Stream has a specific StreamID, as that would imply knowing fully the data that would be created a-priory.  It seems more likely that we'd have capabilities that allow creating a stream so long as it has specific metadata properties, but it's fine not to worry about that for now.  So I think leaving StreamID optional and not passing it in `_applyGenesis` is fine, but it would be good to explain more about what the StreamID is used for and when it might be missing in the comments for `_verifySignature`

Reply:
> I doubt we'll have capabilities that grant the right to create a new Stream

We do, but not in this first iteration. I think passing the streamId from genesis should not cause any issues for now either though.


## For context from the spec:

### Format: Capability

For capability representation on IPFS we use [CACAO](https://github.com/ChainAgnostic/CAIPs/pull/74).

Caveats in CACAO are represented as URIs. We are interested in the following caveats:

- write to stream: `ceramic://<stream-id>`
- write to stream payload X, identified by `<payload-cid>` here - this gives a capability owner a permission to update a stream once: `ceramic://<stream-id>?payload=<payload-cid>`
- create a stream: `ceramic://*?payload=<payload-cid>`, where `payload-cid` is actual *payload* for non-deterministic stream genesis commit.
- Make sure that `ceramic://*` (without *payload* query param) is explicitly **not** allowed